### PR TITLE
Implement gapi signin

### LINKED ISSUE #13: Update BigQueryService to use GAPI

### LINKED ISSUE #15: Implement Google API (gapi) to restrict access to the app

### PR DESCRIPTION
**Pull request by ndsagaga**
Closes #15 
Closes #13 

Implemented Google API to restrict access to authenticated users.

Future: 
- Use gapi to get Oauth token and run BigQueery jobs

### CODE REVIEW

File: web/scm/src/app/services/gapi/GoogleApiService.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+export class GoogleApiService {
+    private readonly gapiUrl: string = 'https://apis.google.com/js/api.js';
+    private config: GoogleApiConfig;
+
+    constructor() {
+        this.config = new GoogleApiConfig({
+            client_id: environment.clientId,
+            discoveryDocs: [environment.bigQueryDiscoveryDocument],
+            scope: [
+                'https://www.googleapis.com/auth/bigquery.readonly'
```
Comment (dpronin) [inline web/scm/src/app/services/gapi/GoogleApiService.ts:18]: please move to contstants

File: web/scm/src/app/services/gapi/GoogleApiService.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    public onLoad(): Observable<boolean> {
+        return this.loadGapi();
+    }
+
+    public getConfig(): GoogleApiConfig {
+        return this.config;
+    }
+
+    private loadGapi(): Observable<boolean> {
```
Comment (dpronin) [inline web/scm/src/app/services/gapi/GoogleApiService.ts:32]: please add a documentation

File: web/scm/src/app/services/gapi/GoogleAuthService.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        .then(auth => auth.signIn())
+        .then(gUser => {
+            this.googleUser = gUser;
+            const authResponse = this.googleUser.getAuthResponse();
+            sessionStorage.setItem(
+                GoogleAuthService.OAUTH_TOKEN_STORAGE_KEY, authResponse.access_token
+            );
+            sessionStorage.setItem(
+                GoogleAuthService.OAUTH_EXPIRY_STORAGE_KEY, authResponse.expires_at.toString()
+            );
```
Comment (dpronin) [inline web/scm/src/app/services/gapi/GoogleAuthService.ts:64]: please a TODO to migrate to async/await. I think we need to maintain one async approach and maybe later we do refactoring to have async/await only. WDYT?

File: web/scm/src/app/services/gapi/GoogleApiConfig.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import ClientConfig = gapi.auth2.ClientConfig;
+
+export interface NgGapiClientConfig extends ClientConfig {
```
Comment (dpronin) [inline web/scm/src/app/services/gapi/GoogleApiConfig.ts:3]: Maybe we just can go with GapiClientConfig?

Comment (ndsagaga) [reply web/scm/src/app/services/gapi/GoogleApiConfig.ts:3]: Done

File: web/scm/src/app/services/gapi/GoogleApiService.ts (hunk 1 above)
Comment (ndsagaga) [reply web/scm/src/app/services/gapi/GoogleApiService.ts:18]: Done

File: web/scm/src/app/services/gapi/GoogleAuthService.ts (hunk 3 above)
Comment (ndsagaga) [reply web/scm/src/app/services/gapi/GoogleAuthService.ts:64]: I initially did have the async/await technique. However, the promise returned by the GoogleAuth object is a reference to itself, which throws an error. 

Screen Shot 2020-05-28 at 7 20 23 PM

**Pull request merged**